src/pytest_cython/plugin.py

```diff
@@ -37,193 +32,128 @@ def pytest_addoption(parser):
+        return
 
-def check_python_versions():
-    (major, minor, patch, *_) = sys.version_info
-    if (major == 3):
-        if (minor == 8 and patch < 7) or (minor == 9 and patch < 2):
-            raise RuntimeError(' '.join([
-                "Please update your Python patch version.",
-                "Your current Python version has a known bug w.r.t. EXT_SUFFIX that prevents this plugin from working.",
-            ]))
+    # only run test if matching .so and .pyx files exist
+    return _PatchedDoctestModule.from_parent(parent, path=file_path)
 
 
-def pytest_collect_file(path, parent):
-    cy_exts = ['.py', '.pyx']
+class _PatchedDoctestModule(DoctestModule):
+    def collect(self) -> Iterable[DoctestItem]:
+        mode = ImportMode(self.config.getoption("importmode"))
 
-    config = parent.config
+        if mode is not ImportMode.importlib and IGNORE_IMPORTMISMATCH != "1":
+            # we know we will get an import file mismatch error so ignore it for now,
+            # we will double check the import paths after import.
+            os.environ[IGNORE_IMPORTMISMATCH_KEY] = "1"
 
-    if path.ext in cy_exts and config.getoption('--doctest-cython'):
-        bin_path = path.new(ext=EXT_SUFFIX)
-        bin_check = bin_path.check()
-        check_python_versions()
-
-        if bin_check:
-            # only run test if matching .so and .pyx files exist
-            # create addoption for this ??
-            if hasattr(DoctestModule, 'from_parent'):
-                try:
-                    # fspath for Node constructors deprecated since
-                    # pytest version 7.0., replaced with pathlib.Path
-                    return DoctestModule.from_parent(parent, path=pathlib.Path(str(path)))
-                except TypeError:
-                    return DoctestModule.from_parent(parent, fspath=path)
+        try:
+            items = list(super().collect())  # import the module and collect tests
+        finally:
+            # set ignore variable back to its original value
+            os.environ[IGNORE_IMPORTMISMATCH_KEY] = IGNORE_IMPORTMISMATCH
+
+        module = self.obj  # module already imported
+
+        try:
+            _check_module_import(module, self.path, mode)
+        except Collector.CollectError:
+            if self.config.getvalue("doctest_ignore_import_errors"):
+                skip("unable to import module %r" % self.path)
             else:
-                # Backwards-compat for older pytest
-                return DoctestModule(path, parent)
+                raise
 
+        return _add_line_numbers(module, items)
 
-class _PatchedLocalPath(py.path.local):
-    """
-    py.path.local path patched so that py.path.local.pyimport() will work
-    if it is a PEP 3149 ABI extension module.  See _patch_pyimport.
-    """
 
-    @property
-    def purebasename(self):
-        return super().basename.replace(EXT_SUFFIX, '')
+def _without_suffixes(path: str | pathlib.Path) -> pathlib.Path:
```

Comment:
Hi, the use of `|` for the type annotation breaks code using this plugin on Python 3.8 and 3.9 (both still supported by pytest), see https://docs.python.org/3/library/typing.html#typing.Union